AllASPFuncs.py

```diff
@@ -1194,14 +1212,19 @@ def build_lightcurve(ID, exposures, sn_path, confusion_metric, flux,
         'host_ra': df['host_ra'][df['id'] == ID].values[0],\
         'host_dec': df['host_dec'][df['id'] == ID].values[0]}
     else:
-        meta_dict = {'ra': df[df['id'] == str(ID)]['ra'].values[0], \
-            'dec': df[df['id'] == str(ID)]['dec'].values[0]}
-
-    data_dict = {'MJD': detections['date'], 'true_flux':
-    detections['realized flux'],  'measured_flux': flux, 'flux_error':
-    sigma_flux}
-    units = {'MJD':u.d, 'true_flux': '',  'measured_flux': '',
-             'flux_error': ''}
+        meta_dict = {'ra': df[df['id'] == str(ID)]['ra'].values[0],
+                     'dec': df[df['id'] == str(ID)]['dec'].values[0]}
```

Comment:
Why are some of these `df["host_ra"][df["id"] == ID]` and some `df[df["id"] == str(ID)]]["host_ra"]`?

In particular, if the second works, I would recommend just extracting that row for `this_one = df[df["id"] == str(ID)]` and then refer to `this_one["ra"].values[0]`, etc.  (Pick a better name than `this_one`).

The idea here is both improved readability, and translating typos into things that you will catch because the variable isn't defined and your linter will catch, instead of a repeated test that you might get wrong and triggers a problem at run time but your linter can't catch.